src/quickgui/plugin/qgsquickplugin.cpp

```diff
@@ -0,0 +1,87 @@
+/***************************************************************************
+  qgsquickplugin.cpp
+  --------------------------------------
+  Date                 : Nov 2017
+  Copyright            : (C) 2017 by Peter Petrik
+  Email                : zilolv at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include <qqml.h>
+
+#include "qgsfeature.h"
+#include "qgsmaplayer.h"
+#include "qgsmessagelog.h"
+#include "qgspointxy.h"
+#include "qgsproject.h"
+#include "qgsrelationmanager.h"
+#include "qgscoordinatetransformcontext.h"
+#include "qgsvectorlayer.h"
+
+#include "qgsquickattributeformmodel.h"
+#include "qgsquickattributeformmodelbase.h"
+#include "qgsquickcoordinatetransformer.h"
+#include "qgsquickfeaturemodel.h"
+#include "qgsquickfeaturehighlight.h"
+#include "qgsquickidentifykit.h"
+#include "qgsquickfeature.h"
+#include "qgsquickmapcanvasmap.h"
+#include "qgsquickmapsettings.h"
+#include "qgsquickmaptransform.h"
+#include "qgsquickmessagelogmodel.h"
+#include "qgsquickplugin.h"
+#include "qgsquickpositionkit.h"
+#include "qgsquickscalebarkit.h"
+#include "qgsquicksubmodel.h"
+#include "qgsquickutils.h"
+
+static QObject *_utilsProvider( QQmlEngine *engine, QJSEngine *scriptEngine )
+{
+  Q_UNUSED( engine )
+  Q_UNUSED( scriptEngine )
+  return QgsQuickUtils::instance();  // the object will be owned by QML engine and destroyed by the engine on exit
+}
+
+void QgisQuickPlugin::registerTypes( const char *uri )
+{
+  qDebug( "REGISTERING QQmlExtensionInterface: QgisQuick" );
+
+  qRegisterMetaType< QList<QgsMapLayer *> >( "QList<QgsMapLayer*>" );
+  qRegisterMetaType< QgsAttributes > ( "QgsAttributes" );
+  qRegisterMetaType< QgsCoordinateReferenceSystem >( "QgsCoordinateReferenceSystem" );
+  qRegisterMetaType< QgsCoordinateTransformContext >( "QgsCoordinateTransformContext" );
+  qRegisterMetaType< QgsFeature > ( "QgsFeature " );
+  qRegisterMetaType< QgsFeatureId > ( "QgsFeatureId" );
+  qRegisterMetaType< QgsPoint >( "QgsPoint" );
+  qRegisterMetaType< QgsPointXY >( "QgsPointXY" );
+  qRegisterMetaType< QgsQuickFeature >( "QgsQuickFeature" );
+
+  qmlRegisterType< QgsProject >( uri, 0, 1, "Project" );
```

Comment:
Should the version reflect the QGIS version? This would probably reduce confusion among developers.

Reply:
I think that the API is not stable yet and will change a lot in following versions. Also, QT has a different version for QtQuickComponents and Qt core. I would stick with 0.1 version and use QGIS version when we stabilize API.

Reply:
Will we continue to ship old versions of the API (as in Qt where you can also import older ones with newer releases)?

Reply:
No for version 0.1. I would stick with 0.1-Experimental until we have something stable. I would say that mayber from QGIS 3.4 (LTR) we should move to versioning based on QGIS ...